src/praisonai/praisonai/integrations/docker_managed_agent.py

```
@@ -0,0 +1,308 @@
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc):
+        self.shutdown()
+
+    # ── the work ─────────────────────────────────────────────────────────────
+    def _execute_sync(self, prompt: str, **kwargs) -> str:
+        self._ensure_container()
+        payload = json.dumps({"agent": self._config, "prompt": prompt})
+
+        _write_file(self._container, "/tmp/praison_agent.json", payload)
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/praison_agent.json"

(S108)